### ISSUE TITLE
Build fails on arm64 alpine:3.8 gcc 8.2.0

### ISSUE THREAD

**Issue by adamkdean** [lovell/farmhash#24]
Installing farmhash on an arm64 machine inside a Ubuntu docker image works fine.

```
prebuild-install WARN install No prebuilt binaries found (target=8.10.0 runtime=node arch=arm64 platform=linux)
make: Entering directory '/app/node_modules/farmhash/build'
  CXX(target) Release/obj.target/farmhash/src/upstream/farmhash.o
  CXX(target) Release/obj.target/farmhash/src/bindings.o
  SOLINK_MODULE(target) Release/obj.target/farmhash.node
  COPY Release/farmhash.node
make: Leaving directory '/app/node_modules/farmhash/build'
```

However, attempting to install it in an Alpine image (Alpine 3.8 with gcc 8.2.0) fails:

```
prebuild-install WARN install No prebuilt binaries found (target=10.14.1 runtime=node arch=arm64 platform=linux)
make: Entering directory '/app/node_modules/farmhash/build'
  CXX(target) Release/obj.target/farmhash/src/upstream/farmhash.o
cc1plus: error: unknown value 'armv8-\xd0\x1b88' for -march
cc1plus: note: valid arguments are: armv8-a armv8.1-a armv8.2-a armv8.3-a armv8.4-a native; did you mean 'armv8-a'?
make: *** [farmhash.target.mk:112: Release/obj.target/farmhash/src/upstream/farmhash.o] Error 1
make: Leaving directory '/app/node_modules/farmhash/build'
```

Attempting to override -march with CFLAGS and CXXFLAGS:

```
prebuild-install WARN install No prebuilt binaries found (target=10.14.1 runtime=node arch=arm64 platform=linux)
make: Entering directory '/app/node_modules/farmhash/build'
  CXX(target) Release/obj.target/farmhash/src/upstream/farmhash.o
cc1plus: error: unknown value 'armv8-a-march=armv8-a' for -march
cc1plus: note: valid arguments are: armv8-a armv8.1-a armv8.2-a armv8.3-a armv8.4-a native
make: *** [farmhash.target.mk:112: Release/obj.target/farmhash/src/upstream/farmhash.o] Error 1
make: Leaving directory '/app/node_modules/farmhash/build'
```

It appears to ignore the values provided and concocts its own bizarre string `armv8-a-march=armv8-a` which seems strange. It's almost as if it's ignoring the CFLAGS and CXXFLAGS.

Does anyone have any ideas?

**Comment by lovell** [comment]
My best guess would be that there's an unterminated string problem in gcc's CPU detection logic when it tries to replace `-march=native` with the actual architecture on ARMv8.

We should probably add `arm64` to the exclusions list for this at https://github.com/lovell/farmhash/blob/master/binding.gyp#L28 - happy to accept a PR if you're able.

**Comment by lovell** [comment]
This has been addressed in v3.0.0, which now only sets `-march=native` for ia32 and x64 targets.

**Comment by adamkdean** [comment]
Thanks for the update @lovell
